### ISSUE TITLE
UnsupportedOperationException on .removeTS3Listeners

### ISSUE THREAD

**Issue by greg2010** [TheHolyWaffle/TeamSpeak-3-Java-API#294]
Hello,

I'm trying to remove an event listener registered via `TS3ApiAsync.addTS3Listeners` and getting `UnsupportedOperationException`.
Stacktrace:
```
java.lang.UnsupportedOperationException: null
	at java.util.concurrent.CopyOnWriteArrayList$COWIterator.remove(CopyOnWriteArrayList.java:1178)
	at com.github.theholywaffle.teamspeak3.EventManager.removeListeners(EventManager.java:72)
	at com.github.theholywaffle.teamspeak3.TS3ApiAsync.removeTS3Listeners(TS3ApiAsync.java:4146)
	at org.kys.cerberus.clients.TeamspeakClient$JoinListener.$anonfun$onClientJoin$3(TeamspeakClient.scala:71)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:654)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:288)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run$$$capture(Promise.scala:60)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```
I've done cursory search and it seems like the `tasks` array is backed by Java's `CopyOnWriteArrayList`. The iterator in `CopyOnWriteArrayList` does not implement `.remove`, and so the exception is thrown.
Javadoc: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CopyOnWriteArrayList.html#iterator--
> The iterator does NOT support the remove method.

**Comment by rogermb** [comment]
Yup, that's a bug. I'll get this fixed for the next release. Also, thanks for doing the research on the Iterator's remove method.

It's kind of amazing that this has been broken for 1.5 years and you're the first one to notice (and report) this bug 😄 

As a workaround for the time being, you could probably just add an `enabled` boolean to your listeners 😃

**Comment by essemX** [comment]
guess i'm the second to notice this, 8 months after the fix still not in a stable release ¯\_(ツ)_/¯

**Comment by rogermb** [comment]
Fixed in 4c3420519b5be992370150a8dd77da049065c24f and finally released in [v1.3.0](https://github.com/TheHolyWaffle/TeamSpeak-3-Java-API/releases/tag/v1.3.0) 🎉
